src/hierarchies/likelihoods/states/multi_ls_state.h

```diff
@@ -0,0 +1,93 @@
+#ifndef BAYESMIX_HIERARCHIES_LIKELIHOODS_STATES_MULTI_LS_STATE_H_
+#define BAYESMIX_HIERARCHIES_LIKELIHOODS_STATES_MULTI_LS_STATE_H_
+
+#include <stan/math/rev.hpp>
+#include <tuple>
+
+#include "algorithm_state.pb.h"
+#include "src/utils/proto_utils.h"
+
+namespace State {
+
+template <typename T>
+Eigen::Matrix<T, Eigen::Dynamic, 1> multi_ls_to_unconstrained(
+    Eigen::Matrix<T, Eigen::Dynamic, 1> mean_in,
+    Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> prec_in) {
+  Eigen::Matrix<T, Eigen::Dynamic, 1> prec_out =
+      stan::math::cov_matrix_free(prec_in);
+  Eigen::Matrix<T, Eigen::Dynamic, 1> out(mean_in.size() + prec_out.size());
+  out << mean_in, prec_out;
+  return out;
+}
+
+template <typename T>
+std::tuple<Eigen::Matrix<T, Eigen::Dynamic, 1>,
+           Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>>
+multi_ls_to_constrained(Eigen::Matrix<T, Eigen::Dynamic, 1> in) {
+  double dim_ = 0.5 * (std::sqrt(8 * in.size() + 9) - 3);
+  double dimf;
+  assert(modf(dim_, &dimf) == 0.0);
+  int dim = int(dimf);
+  Eigen::Matrix<T, Eigen::Dynamic, 1> mean(dim);
+  mean << in.head(dim);
+  Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> prec(dim, dim);
+  prec = stan::math::cov_matrix_constrain(in.tail(in.size() - dim), dim);
+  return std::make_tuple(mean, prec);
+}
+
+// SEE GitHub for tests
```

Comment:
? Remove, I guess?

Reply:
@mberaha and I were discussing about some tests to improve the computation of ``log_det_jac``. I am leaving this conversation open as a reminder